app.rb

```diff
@@ -0,0 +1,59 @@
+require "./lib/chitter"
+require 'sinatra/base'
+require 'sinatra/flash'
+require "./lib/users"
+
+class ChitterManager < Sinatra::Base
+
+  enable :method_override
+  enable :sessions
+
+  register Sinatra::Flash
+
+  get '/' do
+    erb :sign_in
+  end
+
+  post '/sign_in' do
+    session[:username] = params[:username]
```

Comment:
This feels like quite a long method - worth thinking about ways you can refactor it to make it more readable, either by moving code into named methods or thinking about whether some of the code actually belongs in another class.  For example, checking whether the user is duplicate or not is probably not the responsibility of this class, you could put that in a Users.register class method, and as part of that validate whether the information is unique.